### PR TITLE
4750 wealth quintiles

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

Migrated:
- Wealth quintiles service 
- The change listener for dbsync
- Unit tests of wealth quintiles and app component

Refactored services imports and app component init.

https://github.com/medic/angular10-migration/issues/57
https://github.com/medic/angular10-migration/issues/43

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@dianabarsan Hi! when you have time can you please take a look? I think there's still code to migrate in app-component, I can maybe migrate that in another PR 🤔
Thank you!

**Review by dianabarsan — APPROVED** [review]
Great work! Thanks! 
I just have a couple of nits about naming things.

File: webapp/src/ts/services/wealth-quintiles-watcher.service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const subscription = this.changesService.subscribe({
+      key: 'wealth-quintiles',
+      filter: (change:any) => !!(change?.doc?.fields && 'NationalQuintile' in change.doc.fields),
+      callback: (change:any) => this.updateDocs(change)
+    });
+    this.subscriptions.add(subscription);
+  }
+
+  private updateDocs(change:any) {
+    const changedFields = change.doc.fields;
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/services/wealth-quintiles-watcher.service.ts]: Done!

File: webapp/src/ts/app.component.ts [hunk 2]
```diff
@@ -284,27 +254,54 @@ export class AppComponent implements OnInit, OnDestroy {
...
+    if (this.androidAppVersion) {
+      this.authService
+        .has('can_log_out_on_android')
+        .then(canLogout => this.canLogOut = canLogout);
+    } else {
+      this.canLogOut = true;
+    }
+  }
+
+  private persistStorage() {
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/app.component.ts]: Done!

File: webapp/src/ts/app.component.ts [hunk 3]
```diff
@@ -230,49 +236,13 @@ export class AppComponent implements OnInit, OnDestroy {
...
-    if (typeof window.medicmobile_android?.getAppVersion === 'function') {
-      this.globalActions.setAndroidAppVersion(window.medicmobile_android.getAppVersion());
-    }
-
-    if (this.androidAppVersion) {
-      this.authService.has('can_log_out_on_android').then(canLogout => this.canLogOut = canLogout);
-    } else {
-      this.canLogOut = true;
-    }
+    this.startWealthQuintiles();
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/app.component.ts]: Done!

File: webapp/src/ts/app.component.ts [hunk 4]
```diff
@@ -324,7 +321,9 @@ export class AppComponent implements OnInit, OnDestroy {
         }
       },
     });
+  }
 
+  private watchChangesInboxUserContext() {
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/app.component.ts]: Done!

File: webapp/src/ts/app.component.ts [hunk 5]
```diff
@@ -339,25 +338,61 @@ export class AppComponent implements OnInit, OnDestroy {
         this.sessionService.init().then(refresh => refresh && this.showUpdateReady());
       },
     });
+  }
 
+  private watchChangesInboxTranslations() {
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/app.component.ts]: Done!

File: webapp/tests/karma/unit/controllers/inbox.js [hunk 6]
```diff
@@ -134,12 +135,6 @@ describe('InboxCtrl controller', () => {
     });
   });
 
-  it('should cancel the relative date update recurring process when destroyed', () => {
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/tests/karma/unit/controllers/inbox.js:137]: Done!

File: webapp/tests/karma/ts/services/wealth-quintiles-watcher.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+        {
+          doc: {
+            _id: '1',
+            place_id: '123',
+            wealth_quintile_national: 50,
+            wealth_quintile_urban: 50
+          }
+        },
+        {
+          doc: {
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/tests/karma/ts/services/wealth-quintiles-watcher.service.spec.ts:200]: Done! I added an extra doc in another test (above), I'd like this one to bulk empty array.
